packages/babel-plugin/src/styled/__tests__/behaviour.test.tsx

```diff
@@ -463,19 +512,49 @@ describe('styled component behaviour', () => {
     );
   });
 
+  it('should not allow a logical statement with a conditional right-hand side', () => {
+    expect(() =>
+      transform(`
+      import { styled } from '@compiled/react';
+
+      const Component = styled.div\`
+        \${props => props.isShown && (props.isPrimary ? { color: 'blue' } : { color: 'green' })};
+      \`;
+    `)
+    ).toThrow("ConditionalExpression isn't a supported CSS type");
+  });
+
   it('should apply conditional CSS when using "key: value" in string form', () => {
     const actual = transform(`
       import { styled } from '@compiled/react';
 
       const Component = styled.div\`
-        \${props => props.isPrimary ?  'color: green' : \`color: red\`};
+        \${props => props.isPrimary ? 'color: green' : \`color: red\`};
+      \`;
+    `);
+
+    expect(actual).toIncludeMultiple([
+      '._syazbf54{color:green}',
+      '._syaz5scu{color:red}',
+      'className={ax(["",props.isPrimary?"_syazbf54":"_syaz5scu",props.className])}',
+    ]);
+  });
+
+  it('should apply conditional CSS when using "key: value; key: value; ..." in string form', () => {
```

Comment:
hmm this is cool, previously consumer needed to use `css()` to do multi style declarations. I wonder if styled-components supports this?

Reply:
Ah yeah, I didn't do anything particular to support this. It already worked before my changes, it would just be pulled out into logical statements like...
```
props.isPrimary&&"_syazbf54 _1wyb1fwx", !props.isPrimary&&"_syaz5scu _1wybexct"
```

Reply:
Looks like this would work in Styled Components as well, based on the little demo they have:
<img width="640" alt="Screen Shot 2021-11-11 at 11 42 25 am" src="https://user-images.githubusercontent.com/860852/141216311-f0aab627-053f-4b83-ab55-ff12a3872607.png">

Reply:
Yep, using the demo on their site, I can confirm this also works in styled-components.